### ISSUE TITLE
TWAMP tool returning bogus time(s)

### ISSUE THREAD

**Issue by mfeit-internet2** [perfsonar/pscheduler#746]
Seen while testing changes for #741:

```
pscheduler task --tool twping rtt --protocol twamp --dest ps-dev-el7-1.es.net
Submitting task...
Task URL:
https://localhost/pscheduler/tasks/f0b2c3c4-3e38-4dee-be4a-3ea47f7ff3f6
Running with tool 'twping'
Fetching first run...

Next scheduled run:
https://localhost/pscheduler/tasks/f0b2c3c4-3e38-4dee-be4a-3ea47f7ff3f6/runs/9e1c487c-c0f4-4b4a-9439-6f5fadd42ded
Starts 2018-12-17T14:33:41-05:00 (~8 seconds)
Ends   2018-12-17T14:34:02-05:00 (~20 seconds)
Waiting for result...

Problem fetching results: Failed to format result: At /max: '-P43449DT19H33M43.012532S' is not a valid ISO 8601 duration.

No further runs scheduled.

```

**Comment by mfeit-internet2** [comment]
Looking at the data produced by the tool, it looks like the root cause of this is bogus RTTs.

**Comment by mfeit-internet2** [comment]
This is what's coming back from `/usr/bin/twping -R -c 3 -i 0.1 -L 0 '[ps-dev-el7-1.es.net]:862'` when the source end isn't synchronized (split and annotated):
```
FORWARD:

 0 0                            Seq
 1 16143227105333012747         Src Timestamp
 2 0                            Src Synchronized
 3 2.32831e-10                  Src Error
 4 00000000000000000000         Dest Timestamp
 5 1                            Dest Synchronized
 6 0.000139236                  Dest Error
 7 255                          TTL

REVERSE:

 8 0                            Seq
 9 00000000000000000000         Src Timestamp
10 1                            Src Synchronized
11 0.000139236                  Src Error
12 00000000000000000000         Dest Timestamp
13 1                            Dest Synchronized
14 0.000139236                  Dest Error
15 255                          TTL  
```

I think the plugin needs to either this a failure when it happens, or we need to make twping produce correct values even when unsynchronized.

**Comment by vvidic** [comment]
AFAICT this is dst - src timestamp (00000000000000000000 - 16143227105333012747) and this happens when the reply packets get dropped for some reason:

```
2 sent, 2 lost (100.000%), 0 send duplicates, 0 reflect duplicates
```

**Comment by vvidic** [comment]
I can reproduce as follows:

```
$ pscheduler task --tool twping rtt --protocol twamp --dest pstest2.geant.carnet.hr
Submitting task...
Task URL:
https://localhost/pscheduler/tasks/b70e956a-3fa5-4036-853f-42fc6a3dd81e
Running with tool 'twping'
Fetching first run...

Next scheduled run:
https://localhost/pscheduler/tasks/b70e956a-3fa5-4036-853f-42fc6a3dd81e/runs/8eafd553-63b0-4651-898d-0cdfff821bdf
Starts 2019-02-13T11:25:13+01:00 (~8 seconds)
Ends   2019-02-13T11:25:34+01:00 (~20 seconds)
Waiting for result...

Problem fetching results: Failed to format result: At /max: '-P43507DT10H25M14.374576S' is not a valid ISO 8601 duration.

No further runs scheduled.
```
```
$ twping  pstest2.geant.carnet.hr
twping: FILE=time.c, LINE=112, NTP: Status UNSYNC (clock offset issues likely)
Approximately 13.0 seconds until results available

--- twping statistics from [stretch]:9189 to [pstest2.geant.carnet.hr]:19094 ---
SID:    c1c6f11ee00e7189986765a62ea191a2
first:  2019-02-13T11:35:54.682
last:   2019-02-13T11:36:06.114
7 sent, 7 lost (100.000%), 0 send duplicates, 0 reflect duplicates
round-trip time min/median/max = nan/nan/nan ms, (unsync)
send time min/median/max = nan/nan/nan ms, (unsync)
reflect time min/median/max = nan/nan/nan ms, (unsync)
reflector processing time min/max = nan/nan ms
two-way jitter = nan ms (P95-P50)
send jitter = nan ms (P95-P50)
reflect jitter = nan ms (P95-P50)
send TTL not reported
reflect TTL not reported
```

```
$ twping -R pstest2.geant.carnet.hr
twping: FILE=time.c, LINE=112, NTP: Status UNSYNC (clock offset issues likely)
0 16144966644408100879 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255 0 00000000000000000000 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255
1 16144966644757883241 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255 0 00000000000000000000 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255
21 16144966653419564580 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255 0 00000000000000000000 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255
39 16144966661737937728 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255 0 00000000000000000000 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255
52 16144966671242037706 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255 0 00000000000000000000 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255
71 16144966679447040978 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255 0 00000000000000000000 0 2.32831e-10 00000000000000000000 0 2.32831e-10 255
```

**Comment by vvidic** [comment]
After the patch same run looks like this:

```
$ pscheduler task --tool twping rtt --protocol twamp --dest pstest2.geant.carnet.hr
Submitting task...
Task URL:
https://localhost/pscheduler/tasks/7b9925ea-b3e4-424a-a5e2-075326dc5b4e
Running with tool 'twping'
Fetching first run...

Next scheduled run:
https://localhost/pscheduler/tasks/7b9925ea-b3e4-424a-a5e2-075326dc5b4e/runs/6649bb0e-7043-442c-bfb0-664861bd6ea2
Starts 2019-02-13T12:18:44+01:00 (~7 seconds)
Ends   2019-02-13T12:19:05+01:00 (~20 seconds)
Waiting for result...


100% Packet Loss  RTT Min/Mean/Max/StdDev = Unknown/Unknown/Unknown/Unknown ms

No further runs scheduled.
```

**Comment by vvidic** [comment]
PR merged for 4.1.6 release, closing.
